### PR TITLE
add hkust tdnn and update results

### PR DESCRIPTION
**Pull request by naxingyu**

### CODE REVIEW

File: egs/swbd/s5c/RESULTS [hunk 1]
```diff
@@ -142,7 +142,3 @@ exit 0
 %WER 14.6 | 1831 21395 | 87.1 9.2 3.6 1.8 14.6 51.8 | exp/tri7uc-sat_mmi_b0.1/decode_eval2000_graph_sw1_tg_it4/score_17_0.0/eval2000.ctm.swbd.filt.sys
 # fisher 4gram rescoring,
 %WER 13.2 | 1831 21395 | 88.3 8.2 3.4 1.5 13.2 49.2 | exp/tri7uc-sat_mmi_b0.1/decode_eval2000_graph_sw1_fsh_fg_it4/score_19_0.0/eval2000.ctm.swbd.filt.sys
-
-# nnet3 LSTM recipe local/nnet3/run_lstm.sh (30.09.2015), swbd subset
-%WER 16.8 | 1831 21395 | 84.8 10.4 4.7 1.6 16.8 54.3 | exp/nnet3/lstm_ld5/decode_eval2000_sw1_tg/score_10_0.0/eval2000.ctm.swbd.filt.sys
-%WER 15.7 | 1831 21395 | 85.6 9.6 4.8 1.3 15.7 52.8 | exp/nnet3/lstm_ld5/decode_eval2000_sw1_fsh_fg/score_12_0.0/eval2000.ctm.swbd.filt.sys
```
Comment (danpovey) [inline egs/swbd/s5c/RESULTS]: @vijayaditya, do you have any better numbers than this?
I'll merge this anyway, but this is a big disappointing.

Comment (vijayaditya) [inline egs/swbd/s5c/RESULTS]: # these are results with nnet3 LSTMs : local/nnet3/run_lstm.sh (8 epoch training on speed-perturbed and volume perturbed data)
 %WER 11.8 | 1831 21395 | 89.4 7.0 3.6 1.3 11.8 47.8 | exp/nnet3/lstm_ld5_sp/decode_eval2000_sw1_fsh_fg/score_13_0.0/eval2000_hires.ctm.swbd.filt.sys
 %WER 13.2 | 1831 21395 | 88.3 8.0 3.7 1.5 13.2 49.5 | exp/nnet3/lstm_ld5_sp/decode_eval2000_sw1_tg/score_11_0.0/eval2000_hires.ctm.swbd.filt.sys
 
 # results with nnet3 tdnn: local/nnet3/run_tdnn.sh (11.10.2015) (2 epoch training on speed-perturbed and volume perturbed data)
 %WER 12.1 | 1831 21395 | 89.1 7.1 3.8 1.3 12.1 48.1 | exp/nnet3/tdnn_sp/decode_eval2000_hires_sw1_fsh_fg/score_12_0.0/eval2000_hires.ctm.swbd.filt.sys
 %WER 13.6 | 1831 21395 | 87.9 8.2 3.9 1.5 13.6 51.0 | exp/nnet3/tdnn_sp/decode_eval2000_hires_sw1_tg/score_11_0.0/eval2000_hires.ctm.swbd.filt.sys

Comment (danpovey) [inline egs/swbd/s5c/RESULTS]: Thanks- perhaps you could create a pull request for this, including the
scripts?
Dan

Comment (vijayaditya) [inline egs/swbd/s5c/RESULTS]: These are already checked in both master and ctc branches.

Comment (danpovey) [inline egs/swbd/s5c/RESULTS]: Sorry, I misunderstood the patch- he was deleting out-of-date results, not adding them.

File: egs/hkust/s5/local/nnet3/run_ivector_common.sh [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  # this one has a fairly small dim (defaults to 100)
+  steps/online/nnet2/train_ivector_extractor.sh --cmd "$train_cmd" --nj 10 \
+    data/train_hires exp/nnet3/diag_ubm exp/nnet3/extractor || exit 1;
+fi
+
+if [ $stage -le 5 ] && [ "$speed_perturb" == "true" ]; then
+  # Although the nnet will be trained by high resolution data,
+  # we still have to perturbe the normal data to get the alignment
+  # _sp stands for speed-perturbed
+  utils/perturb_data_dir_speed.sh 0.9 data/train data/temp1
```
Comment (vijayaditya) [inline egs/hkust/s5/local/nnet3/run_ivector_common.sh]: @naxingyu this patch copies my mistakes in SWBD recipe.. please provide data/train_hires to perturb_data_dir.sh script. This is the directory with volume perturbation.

Comment (vijayaditya) [inline egs/hkust/s5/local/nnet3/run_ivector_common.sh]: I corrected the bug in SWBD recipe. Please check the pull request #251 . I am using non volume perturbed data for _sp alignment generation and volume perturbed data for ivector extraction and lda+mllt estimation.

Comment (naxingyu) [inline egs/hkust/s5/local/nnet3/run_ivector_common.sh]: According to the comments, normal data (not hires) is only used to 
generate alignments, the actual perturbed hires mfcc is extracted later, 
after aligning.
Or the comment is misleading?

File: egs/hkust/s5/local/nnet3/run_ivector_common.sh [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    steps/make_mfcc.sh --cmd "$train_cmd" --nj 70 \
+      data/$x exp/make_mfcc/$x $mfccdir || exit 1;
+    steps/compute_cmvn_stats.sh data/$x exp/make_mfcc/$x $mfccdir || exit 1;
+  done
+  utils/fix_data_dir.sh data/train_sp
+
+  $align_script --nj 30 --cmd "$train_cmd" \
+    data/train_sp data/lang $gmm_dir ${gmm_dir}_sp_ali || exit 1
+
+  # Now perturb the high resolution daa
```
Comment (naxingyu) [inline egs/hkust/s5/local/nnet3/run_ivector_common.sh]: Maybe this comment should be
# Now generate high resolution data from perturbed data

File: egs/hkust/s5/local/nnet3/run_ivector_common.sh (hunk 2 above)
Comment (naxingyu) [inline egs/hkust/s5/local/nnet3/run_ivector_common.sh]: Now I get it. The trick is to use both -vol and -speed when calling sox. 
Thanks.

**Pull request merged**